dsperse/src/analyzers/runner_analyzer.py

```
@@ -229,18 +312,216 @@ def _calculate_security(slices):
+        source_path = str(Path(slice_path).resolve())
+        # Check for model-level metadata before attempting normalization/conversion
+        if RunnerAnalyzer._has_model_metadata(Path(slice_path)):
+            p = Path(slice_path)
+            slices_dir = p if (p / 'metadata.json').exists() else (p / 'slices')
+            run_meta = RunnerAnalyzer._build_from_model_metadata(slices_dir)
+            # Determine packaging type from contents of slices_dir
+            try:
+                has_slice_dirs = any(d.is_dir() and d.name.startswith('slice_') for d in slices_dir.iterdir())
+                has_dslice = any(f.is_file() and f.suffix == '.dslice' for f in slices_dir.iterdir())
+                packaging_type = 'dirs' if has_slice_dirs or not has_dslice else 'dslice'
+            except Exception:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)